### PR TITLE
Convert package acceptance tests over to beaker-rspec & serverspec

### PR DESCRIPTION
**Pull request by rodjek**
Pretty much a direct conversion of the existing tests, there's definitely some optimisation that can happen (e.g. the airgapped_usage specs are basically identical to the validate_a_new_module specs, just with additional hooks).

This also introduces a couple of minor monkey patches to beaker-rspec to:
 * specify the working directory that commands should be run in with `let(:cwd)` (expects a string)
 * specify the environment variables that should be set when executing commands with `let(:env)` (expects a hash)

I've also modified the rake task to add a quality of life improvement. I got sick of having to look up the latest build SHA, so it now happens automatically if you set the `SHA` environment variable to `latest`.

If you want to give it a try:
```
cd package-testing; for target in redhat6-64 redhat7-64 sles11-64 sles12-64 ubuntu1404-64 ubuntu1604 osx1011-64 osx1012-64 osx1013-64 windows2012r2-64; do SHA=latest TEST_TARGET=$target bundle exec rake acceptance; done
```

### CODE REVIEW

**Comment by rodjek** [comment]
Apologies in advance for the large diff :crying_cat_face:

File: package-testing/Rakefile [hunk 1]
```diff
@@ -26,6 +27,23 @@ namespace :acceptance do
     # to use this beaker suite outside of Puppet Inc, set BUILD_SERVER to the hostname of a build server that uses the directory hierarchy beaker assumes
     ENV['BUILD_SERVER'] ||= 'builds.delivery.puppetlabs.net'
 
+    if ENV['SHA'] && ENV['SHA'] == 'latest'
+      page = Nokogiri::HTML(open("https://#{ENV['BUILD_SERVER']}/pdk"))
```
Comment (scotje) [inline package-testing/Rakefile]: You can use this url to sort by date descending and avoid needing to sort below: http://builds.delivery.puppetlabs.net/pdk/?C=M&O=D

**Comment by scotje** [comment]
Made one comment (that seems to have been kind of buried) where you could simplify the `SHA=latest` stuff a bit, but other than that this looks great!

File: package-testing/Rakefile (hunk 1 above)
Comment (rodjek) [reply package-testing/Rakefile]: Ah, nice! Will update

**Pull request merged**